### PR TITLE
[HttpClientTest] remove WildcardConnect () on WatchOS

### PR DESCRIPTION
**Pull request by lewurm**
fixes regression mentioned here https://github.com/mono/mono/pull/7358#issuecomment-374678297

### CODE REVIEW

**Comment by lewurm** [comment]
@monojenkins backport 2018-02

**Comment by lewurm** [comment]
@monojenkins rebase

File: mcs/class/System.Net.Http/Test/System.Net.Http/HttpClientTest.cs [hunk 1]
```diff
@@ -1459,6 +1456,7 @@ public void WildcardConnect ()
 				Assert.That (wex.Status, Is.EqualTo (WebExceptionStatus.ConnectFailure), "#3");
```
Comment (akoeplinger) [inline mcs/class/System.Net.Http/Test/System.Net.Http/HttpClientTest.cs]: The proper fix would've been to check that `e.InnerException` is PlatformNotSupportedException.

Comment (lewurm) [reply mcs/class/System.Net.Http/Test/System.Net.Http/HttpClientTest.cs]: @akoeplinger please submit a PR

**Pull request merged**